softlayer/resource_softlayer_global_ip.go

```diff
@@ -0,0 +1,265 @@
+)
+
+func resourceSoftLayerGlobalIp() *schema.Resource {
+        return &schema.Resource{
+                Create:   resourceSoftLayerGlobalIpCreate,
+                Read:     resourceSoftLayerGlobalIpRead,
+                Update:   resourceSoftLayerGlobalIpUpdate,
+                Delete:   resourceSoftLayerGlobalIpDelete,
+                Exists:   resourceSoftLayerGlobalIpExists,
+                Importer: &schema.ResourceImporter{},
+
+                Schema: map[string]*schema.Schema{
+                        "id": &schema.Schema{
+                                Type:     schema.TypeInt,
+                                Computed: true,
+                        },
+
+                        "ip_address": &schema.Schema{
+                                Type:     schema.TypeString,
+                                Computed: true,
+                        },
+
+                        "routes_to": &schema.Schema{
+                                Type:     schema.TypeString,
+                                Required: true,
+                        },
+                },
+        }
+}
+
+func resourceSoftLayerGlobalIpCreate(d *schema.ResourceData, meta interface{}) error {
+        sess := meta.(*session.Session)
+        //destinationIpAddress := d.Get("routes_to").(string)
+
+        // Find price items with AdditionalServicesGlobalIpAddresses
+        productOrderContainer, err := buildGlobalIpProductOrderContainer(d, sess, AdditionalServicesGlobalIpAddressesPackageType)
+        if err != nil {
+                // Find price items with AdditionalServices
+                productOrderContainer, err = buildGlobalIpProductOrderContainer(d, sess, AdditionalServicesPackageType)
+                if err != nil {
+                        return fmt.Errorf("Error creating global ip: %s", err)
+                }
+        }
+
+        log.Println("[INFO] Creating global ip")
+
+        receipt, err := services.GetProductOrderService(sess).
+                PlaceOrder(productOrderContainer, sl.Bool(false))
+        if err != nil {
+                return fmt.Errorf("Error during creation of global ip: %s", err)
+        }
+
+        globalIp, err := findGlobalIpByOrderId(sess, *receipt.OrderId)
+        d.SetId(fmt.Sprintf("%d", *globalIp.Id))
+        return resourceSoftLayerGlobalIpUpdate(d, meta)
+
+}
+
+func resourceSoftLayerGlobalIpRead(d *schema.ResourceData, meta interface{}) error {
+        sess := meta.(*session.Session)
+        service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+        globalIpId, err := strconv.Atoi(d.Id())
+        if err != nil {
+                return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+        }
+
+        globalIp, err := service.Id(globalIpId).Mask(GlobalIpMask).GetObject()
+        if err != nil {
+                if strings.Contains(err.Error(), "404 Not Found") {
+                        d.SetId("")
+                        return nil
+                }
```

Comment:
String comparison is not a reliable way to check for a 404.  Instead, type-assert the error to an `sl.Error` and check the status code directly, for example:

```go
if apiErr, ok := err.(sl.Error); ok && apiErr.StatusCode == 404 {
		d.SetId("")
		return nil
	}
```